File: Python/AddressBuilder.py
from Address import Address
from AddressData import AddressData


class AddressBuilder:
    def __init__(self, address_data: AddressData):
        self.address_Data = address_data
        self.address = Address("")
        self.title_f = ""
        self.title_m = ""

    def get_address(self) -> Address:
        return self.address

    def build_line1(self) -> None:
        title = ""
        if self.address_Data.gender == "m":
            title = self.title_m
        elif self.address_Data.gender == "f":
            title = self.title_f
        # other cases empty string
        self.address.add_line(title + " " + self.address_Data.surname + " " + self.address_Data.name + "\n")

    def build_line2(self) -> None:
        self.address.add_line(self.address_Data.number + " " + self.address_Data.street + "\n")

    def build_line3(self) -> None:
        pass
    def build_line4(self) -> None:
        self.address.add_line(self.address_Data.state)

    def build_line5(self) -> None:
        pass
    def build_line6(self) -> None:
        pass

File: CPP/AddressBuilder.cpp
//
// Created by Nutzer on 22.08.2025.
//

#include "AddressBuilder.h"

#include <iostream>
#include <utility>

void AddressBuilder::buildLine1() {
    std::string title;
    switch(addressData.gender){
        case 'm': title = title_m; break;
        case 'f': title = title_f; break;
            // other cases empty string
    }
    address.addLine(title + " " + addressData.surname + " " + addressData.name + "\n");
}

void AddressBuilder::buildLine2() {
    address.addLine(addressData.number + " " + addressData.street + "\n");
}

void AddressBuilder::buildLine3() {}

void AddressBuilder::buildLine4() {
    address.addLine(addressData.state);
}

void AddressBuilder::buildLine5() {}
void AddressBuilder::buildLine6() {}

Address AddressBuilder::getAddress() {
    return address;
}

AddressBuilder::AddressBuilder(const address::AddressData& address_data)
    :addressData(address_data){}

